packages/page-bounties/src/Bounties.tsx

```diff
@@ -0,0 +1,63 @@
+// Copyright 2017-2020 @polkadot/app-bounties authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useRef } from 'react';
+
+import type { Bytes, Option } from '@polkadot/types';
+import type { BlockNumber, Bounty as BountyType, BountyIndex } from '@polkadot/types/interfaces';
+import { BountyDescription } from '@polkadot/app-bounties/types';
+import { Table } from '@polkadot/react-components';
+import { useApi, useCall } from '@polkadot/react-hooks';
+import { StorageKey } from '@polkadot/types';
+
+import Bounty from './Bounty';
+import { useTranslation } from './translate';
+
+type BountyEntry = [{ args: [BountyIndex] }, Option<BountyType>];
+type BountyDescriptionEntry = [StorageKey, Option<Bytes>];
+
+const transformBountyDescriptions = {
+  transform: (entries: BountyDescriptionEntry[]): BountyDescription[] => {
+    return entries.map(([key, description]) => ({ description: description.value.toHuman() as string, id: key.args[0] }));
+  }
+};
+
+function Bounties (): React.ReactElement {
+  const { t } = useTranslation();
+  const { api } = useApi();
+
+  const bounties = useCall<BountyEntry[]>(api.query.treasury.bounties.entries);
```

Comment:
This is the refresh issue - entries doesn't have subscriptions, so we never know when they update. (There is a Substrate issue logged on this)

Reply:
Darn, thanks. Alternatives as of now?

Reply:
This is where a derive would come in nicely -

- subscribe to the number of bounties
- when changed, update `.keys` (not entries)
- subscribe to each of the keys for updates